.github/workflows/scorecards-analysis.yml

```diff
@@ -0,0 +1,55 @@
+name: Scorecards supply-chain security
+on:
+  # Only the default branch is supported.
+  branch_protection_rule:
+  schedule:
+    - cron: '16 5 * * 3'
+  push:
+    branches: [ master ]
+
+# Declare default permissions as read only.
+permissions: read-all
+
+jobs:
+  analysis:
+    name: Scorecards analysis
+    runs-on: ubuntu-latest
+    permissions:
+      # Needed to upload the results to code-scanning dashboard.
+      security-events: write
```

Comment:
i couldn't figure out what this actually enabled

Reply:
It grants access to the following APIs: https://docs.github.com/en/rest/reference/permissions-required-for-github-apps#permission-on-security-events